types/src/auth.rs

```diff
@@ -0,0 +1,161 @@
+//! types related to accounts
+
+use prost::Message;
+use pbjson_types::Any;
+
+use celestia_proto::cosmos::crypto::ed25519::PubKey as Ed25519PubKey;
+use celestia_proto::cosmos::crypto::secp256k1::PubKey as Secp256k1PubKey;
+use celestia_tendermint::crypto::default::ecdsa_secp256k1::SigningKey;
+use celestia_tendermint::public_key::PublicKey;
+use celestia_tendermint::public_key::Secp256k1 as VerifyingKey;
+use celestia_tendermint_proto::Protobuf;
+
+use crate::Error;
+
+pub use celestia_proto::cosmos::auth::v1beta1::BaseAccount as RawBaseAccount;
+pub use celestia_proto::cosmos::auth::v1beta1::ModuleAccount as RawModuleAccount;
+
+const COSMOS_ED25519_PUBKEY: &str = "/cosmos.crypto.ed25519.PubKey";
+const COSMOS_SECP256K1_PUBKEY: &str = "/cosmos.crypto.secp256k1.PubKey";
+
+/// Params defines the parameters for the auth module.
+#[derive(Debug)]
+pub struct AuthParams {
+    /// Maximum number of memo characters
+    pub max_memo_characters: u64,
+    /// Maximum nubmer of signatures
+    pub tx_sig_limit: u64,
+    /// Cost per transaction byte
+    pub tx_size_cost_per_byte: u64,
+    /// Cost to verify ed25519 signature
+    pub sig_verify_cost_ed25519: u64,
+    /// Cost to verify secp255k1 signature
+    pub sig_verify_cost_secp256k1: u64,
+}
+
+/// [`BaseAccount`] defines a base account type.
+///
+/// It contains all the necessary fields for basic account functionality.
+///
+/// Any custom account type should extend this type for additional functionality
+/// (e.g. vesting).
+#[derive(Debug, Clone, PartialEq)]
+pub struct BaseAccount {
+    /// Bech32 `AccountId` of this account.
+    pub address: String,
```

Comment:
shouldn't we use Address type we have in types?

Reply:
I've missed that we already have it, thanks for the tip. Used that instead :+1: